### DISCUSSION TITLE
Blackbox logging wrong values + Strange behavior.

### DISCUSSION THREAD

**Discussion by Amarcool55** [iNavFlight/inav#7785] [Q&A]
Hi Everyone,
So i have been working on a long range quad with the following config:

- MrCrocHD 6" Frame,
- 6s Battery
- Matek F722-SE
- ZeeZ 50A blHeli32 4in1 ESC 
- flywoo Nin 2305 motors
- TBS Crossfire nano diversity
- Tbs Unify Pro32 Hv
- Matek M9n-5883 GPS and compass

The quad flys good and everything is works fine. i tuned it by logging blackbox data and it is very stable even in windy situations for the last 2 months, except 2 times.
So both of these time when i tried to look at the black box data i found out that blackbox logged wrong debug information. 
This has happened both of these times now and I feel this only happens when there is this strange behaviour where-in the drone just loses all control just like a motor desync and flips over. 

The first time this happened i was flying low and crashed into the ground, But yesterday it recovered almost instantaneously. I now fear to go out long range in-case this happens again and i lose my quad. 

I have attached a video to show the issue and also embedded the backbox viewer overlay on the video.
The stick inputs are recorded correctly but the other things are not correct(Gyro, Battery voltages etc). 
I have also attached the log file here.
I am logging at 50% rate. Please let me what could be wrong here.

Other that these 2 instances the blackbox logs are perfectly recorded and their are no issues in flying as well.
It almost feels that both of these issues are interconnected as whenever the this strange behaviour occurs the blackbox logging goes haywire and i am not able to figure out what could be the real issue behind this.
[LOG00012.TXT](https://github.com/iNavFlight/inav/files/7965501/LOG00012.TXT)

Look at 2:20-2:25 in the video and let me know what you think this looks like.
https://youtu.be/D1c5dkXiDJo

**Comment by stronnag** [comment]
Your log is corrupt. You can easily see this in the header information where a number of field descriptions are missing, fot example the timp stamp is truncated and runs into the `dynamicGyroNotchMinHz` field.
```
H Log start datetime:2022-01-29T14:55:07H dynamicGyroNotchMinHz:120
H acc_lpf_hz:15
```
 The voltage cannot be reliably reported anywhere, as `vbat_scale`  and `vbatref` are missing. In all , more than 40 lines of parameter / setting  data are missing in the header. The SD card cannot sustain reliable logging. The log cannot be trusted.

Try a different card, try a slower logging speed. Perhaps the failed card writes cause device hang resulting in momentary loss of control.

Attached are two files:

* **badlog.txt** : The corrupt / missing part of your corrupt log.
* **goodlog.txt** : The same section of a good log.

[badlog.txt](https://github.com/iNavFlight/inav/files/7965592/badlog.txt)
[goodlog.txt](https://github.com/iNavFlight/inav/files/7965593/goodlog.txt)

**Reply by Amarcool55** [reply]
So you mean this slow SD card is causing the quad to loose control while trying to cope up with the log writing and also corrupt the log too?

**Reply by stronnag** [reply]
I'm telling you your log is corrupt, it cannot be relied upon. I;ve suggested a couple of remedies. The other possible effects of a slow / bad card are speculation consistent with the facts. Only you can investigate further.

**Comment by Amarcool55** [comment]
So, I switched to a faster card and also upgraded to Inav 4.1 but still had a crash today and it was still not logged the log ends abruptly. Do you recommend keeping the blackbox logging on always? i am using a class10 sandisk ultra sd card. @DzikuVx can you please help me with this ?
https://youtu.be/AZSQnUwm8e
[LOG00006.TXT](https://github.com/iNavFlight/inav/files/8056324/LOG00006.TXT)
M
